src/Routing/Router.php

```diff
@@ -988,7 +988,7 @@ public static function parseRoutePath(string $url): array
             $defaults['plugin'] = $matches['plugin'];
         }
         if ($matches['prefix'] !== '') {
-            $defaults['prefix'] = strtolower($matches['prefix']);
+            $defaults['prefix'] = Inflector::underscore($matches['prefix']);
```

Comment:
It's a bit odd that values for `controller` and `plugin` are in CamelCase but `prefix` is underscored even though it maps to namespace part like the other two.

Reply:
I agree CamelCase would be consistent. But it would BC and may be hard to upgrade apps.

There was lowercase prefixes since [1.x](https://book.cakephp.org/1.3/en/The-Manual/Developing-with-CakePHP/Configuration.html#prefix-routing).

This change is to be consistent with `RouteBuilder::prefix()` 
https://github.com/cakephp/cakephp/blob/d68155f315b8fc1bb00ee08df3ba562c1cc51c96/src/Routing/RouteBuilder.php#L866

Reply:
Yeah :disappointed:

Reply:
This has bugged me again and again and confused the heck out of me making me waste a lot of hours in routing setup. 4.x is a good moment to fix that having everything ClassNamy but actions als camelCase.

Reply:
Hehe, same here.
I spend tons of hours also normalizing this for TinyAuth and TinyAuthBackend explaining to me and also to others in docs when it is under_scored and when CamelCased.

We should maybe even think about fixing this before RC for 4.x to be consistent CamelCase always (except for in the URL itself when inflected due to route class).
But should be a follow up to not blocked this being merged now.

Reply:
I agree that if we want to switch to CamelCase for prefixes 4.0.0 is the time to do it. However, it doesn't have to be done in this pull request. It can be done separately before the RC.